### ISSUE TITLE
[BUG] The vocab_size of tokenizer is mismatched with the vocab_size of the model config

### ISSUE THREAD

**Issue by JinchaoLove** [QwenLM/Qwen#553]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

Qwen tokenizer `vocab_size`: 151851
Qwen `vocab_size` in config.json: 151936
What are the additional `85` (151936 - 151851 = 85) tokens used for? Will this cause any unexpected issues?

For example, if we add 100 new tokens in the pretrained tokenizer, the `85` tokens are already in the model's embedding space (**un-trainable** as we do not want to change the LLM's performance), while the new `15` tokens should be added by the model's `resize_token_embeddings()` function (**trainable**).

### 期望行为 | Expected Behavior

The Qwen tokenizer `vocab_size`should be matched with the Qwen `vocab_size` in config.json. Then if we add new tokens, the model's resize_token_embeddings behavior will make sure the new token embeddings are trainable.

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by jklj077** [comment]
> What are the additional 85 (151936 - 151851 = 85) tokens used for?

Please see https://github.com/QwenLM/Qwen/issues/419 and https://github.com/QwenLM/Qwen/issues/482 for our previous responses. `resize_token_embeddings` has an argument named `pad_to_multiple_of` for the exact same reason. 

> Will this cause any unexpected issues?

For one, I don't think transformers support your example natively. (Correct me if I'm wrong.) `resize_token_embeddings` will replace the old embedding module with the new embedding module, which means that it is, as a whole, either trainable or not trainable.

If customized code is needed, you could use the `vocab_size` of the tokenizer to determine which parts needs to be trained. The embedding parameters corresponding to tokens out of the tokenizer are not meaningful (not `in the model's embedding space`) and can be used for new tokens. For example, run `resize_token_embeddings(new_tokenizer_vocab_size, pad_to_multiple_of=128)` and then make sure the first `old_tokenizer_vocab_size` tokens is not trainable.

**Comment by JinchaoLove** [comment]
Thanks for your kind reply. I see, `151851` is the real old vocab size, while `151936` (7B) and `152064` (14B) are for efficient training.
FYI, we can make the whole embedding or lm_head module trainable and then masking out old embeddings' gradients, e.g.,
```python
mask = torch.zeros_like(self.lm_head.weight)
mask[self.config.old_vocab_size:] = 1.
self.lm_head.weight.register_hook(lambda grad: grad*mask)
```

**Comment by szhou0202** [comment]
I am wondering about the same thing. How to resize the model embeddings after adding new special tokens.
